core/services/chainlink/application.go

```diff
@@ -148,16 +148,23 @@ func NewApplication(config *orm.Config, ethClient eth.Client, advisoryLocker pos
 	scryptParams := utils.GetScryptParams(config)
 	keyStore := keystore.New(store.DB, scryptParams)
 
+	telemetryIngressClient := synchronization.TelemetryIngressClient(&synchronization.NoopTelemetryIngressClient{})
 	explorerClient := synchronization.ExplorerClient(&synchronization.NoopExplorerClient{})
 	statsPusher := synchronization.StatsPusher(&synchronization.NoopStatsPusher{})
-	monitoringEndpoint := ocrtypes.MonitoringEndpoint(&telemetry.NoopAgent{})
+	monitoringEndpointGen := telemetry.MonitoringEndpointGenerator(&telemetry.NoopAgent{})
```

Comment:
What does the generator part of this name imply? Sounds like a Factory pattern?

Reply:
The new telemetry ingress client needs to associate telemetry payloads with their contract address. To achieve that, I switched `NewDelegate` in `offchainreporting` from taking a `MonitoringEndpoint` to instead take a function that returns a `MonitoringEndpoint`

So `monitoringEndpointGen` here gets called with the contract address in `NewDelegate` and returns `MonitoringEndpoint`. This address is used by the new telem ingress client, but ignored in the original Explorer and Noop clients (which were updated to also implement this new interface)